eval/builtin_fn.go

```diff
@@ -780,6 +782,57 @@ func rangeFn(ec *EvalCtx, args []Value, opts map[string]Value) {
 	}
 }
 
+func hasValue(ec *EvalCtx, args []Value, opts map[string]Value) {
+	TakeNoOpt(opts)
+
+	var found bool
+	if len(args) == 2 {
+		v := args[0]
+		if iterator, ok := v.(Iterable); ok {
+			iterator.Iterate(func(v Value) bool {
+				if v == args[1] {
+					found = true
+					return false
+				}
+				return true
+			})
+		} else {
+			throw(fmt.Errorf("argument of type '%s' is not iterable", v.Kind()))
+		}
+	} else {
+		throw(errors.New("want 2 arguments"))
+	}
+
+	ec.ports[1].Chan <- Bool(found)
+}
+
+func hasKey(ec *EvalCtx, args []Value, opts map[string]Value) {
+	TakeNoOpt(opts)
+
+	var found bool
+	if len(args) == 2 {
+		v := args[0]
+		switch val := v.(type) {
+		case HasKeyer:
+			if val.HasKey(args[1]) {
+				found = true
+			}
+		case Lener:
```

Comment:
Hmm, this is actually a bit brittle as `Lener` does not always accept numerical indices (e.g. maps also implement `Lener`). But I cannot think of a better solution at the moment, please add a `XXX` comment here like `XXX(xiaq): Not all types that implement Lener have numerical indices`